### ISSUE TITLE
Path::strip_superfluous_slashes() does not strip trailing slash

### ISSUE THREAD

**Issue by ttcoder** [genodelabs/genode#4708]
... which is in contradiction with this comment :-)

``` c
				/* strip superfluous slashes, e.g., "//path/" -> "/path" */
```

Possible actions:
- change the comment as per the code, make it look like ```/* strip superfluous slashes, e.g., "//path/" -> "/path/" */```
- tweak the code as per the comment, to implement trailing slash removal

The latter would mean class Path would be able to do complete normalization, which would be nice.
I took a stab at fixing it with a two-liner (below) but realized it also changes the corner case of the absolute root (single slash absolute path "/"), and reduces it to empty string. So maybe that corner case needs to be special-cased.

``` c++
diff --git a/repos/os/include/os/path.h b/repos/os/include/os/path.h
index 1b929f26e..19ecb93f7 100644
--- a/repos/os/include/os/path.h
+++ b/repos/os/include/os/path.h
@@ -18,6 +18,8 @@
 namespace Genode {
 
        class Path_base;
@@ -102,6 +104,12 @@ class Genode::Path_base
                                /* strip superfluous slashes, e.g., "//path/" -> "/path" */
                                while (path[1] == '/')
                                        remove_char(path);
+                               
+                               if (path[1] == '\0')
+                               {
+                                       remove_char(path);
+                                       break;
+                               }
                        }
```

Context: I ran into this issue, in my "FS indexer" code : using Directory::Path (or maybe a String rather than a Path, gotta check), it creates paths with a trailing slash, sometimes. So I thought I would Path::import() that path into a Path to normalize it, but it does not get normalized.

That's in Genode 22.08 BTW.

**Comment by nfeske** [comment]
I'm sorry for attending this issue so late. Trailing slashes frighten me for some reason.

In fact, in Genode's VFS, the trailing slash of a path has a meaning. It denotes which VFS plugin is responsible for the path. So when passing `/path/to/dir` to the VFS, the plugin that provides the `/path/to` directory dispatches the operation. In contrast, when passing `/path/to/dir/`, the plugin that provides the `/path/to/dir` directory dispatches the operation. The silent removal of trailing slashes throws away this distinction. The ramifications are not entirely clear to me (maybe I'm overly cautious?). Because of this uncertainty, I'm hesitant to incorporate this change.

Fixing the comment is of course a no-brainer (done in a7632616993fe694f6895cd6b58eae9bdb7a218c). Thanks for pointing out this inconsistency. But I guess this does not solve your problem. ;-)

**Comment by ttcoder** [comment]
Actually, we're good! Last week I revisited the code that "spawns" the non-normalized path... And I realized that instead of using path::import(), I can instead use the setter that gets a "pwd" argument, which solves my problem since that latter one does not append a trailing slash, and I no longer need to normalize anything.

Closing as no-need-to-fix :-)

PS. Agreed about the semantics of slashes and what-not, over the years I've grown to be careful about that as well, it's surprising how much head scratching there can be about a supposedly "simple" construct whose building bricks are just dots and slashes.
